### ISSUE TITLE
Unread counts from link collection

### ISSUE THREAD

**Issue by kennyrowe** [urbit/urbit#4191]
**Description:**
Landscape is showing a link collection as bold (unread message) as well as unread counts on the home screen, but there are no unread links in the collection.

```
Ship: ~sicdev-pilnup
Collection Owner: ~sicdev-pilnup
Group: ~dalten/dalten-collective
Hash: 4l8n3
```

**Reproduction:**
- Booted a moon and subscribed to the group and created test links. There were no issues.
- Unable to determine if other subscribed ships are experiencing the same error.

**Screenshots:**
![image](https://user-images.githubusercontent.com/6291612/102646797-879af780-4119-11eb-85a7-7470ff37c810.png)
![image](https://user-images.githubusercontent.com/6291612/102646823-9386b980-4119-11eb-9e99-f3fbbecd8ef7.png)
![image](https://user-images.githubusercontent.com/6291612/102646845-9a153100-4119-11eb-82cc-ee52c70aa962.png)

**Comment by matildepark** [comment]
Phantom unreads seemingly happen predominantly with Links and Publish — is this because their structures both changed producing quirk behaviours? How could this be instigated, and how can we work around and avoid this later? @liam-fitzgerald

**Comment by liam-fitzgerald** [comment]
can you run `:hark-store %foo` and post results?

**Comment by kennyrowe** [comment]
```
[by-index={} ~]
> :hark-store %foo
```

**Comment by urcades** [comment]
Ah, should have found this one, sorry for making the dupe @matildepark: 

@liam-fitzgerald Something I and others have realized is that the "unread state" seem to be triggered via **explicit** clicking of the item in the collection's/notebook's index.

I can imagine a variety of cases where you've either traversed to the item in question via a link in a chat (when we support this more robustly), or when you've used the forward/previous navigation in notebooks to view a next post that don't involve clicking the item at all.

Rather than have 'unread' be triggered onClick, we really should generalize to simply viewing the content in question, being on the page, whatever to make the marking of unread-ness be a lot more generalized.

If we generalize marking 'read' as being based on content visitation, this eliminates a whole host of uncertainties around where to click the item to mark it as read.

**Comment by matildepark** [comment]
Fixed in #4261.
